packages/slider/src/Slider.ts

```diff
@@ -16,44 +16,69 @@ import {
     CSSResultArray,
     TemplateResult,
     query,
-    PropertyValues,
     styleMap,
     ifDefined,
 } from '@spectrum-web-components/base';
-import { streamingListener } from '@spectrum-web-components/base/src/streaming-listener.js';
 
 import sliderStyles from './slider.css.js';
-import { Focusable } from '@spectrum-web-components/shared/src/focusable.js';
 import { StyleInfo } from 'lit-html/directives/style-map';
+import { HandleController } from './HandleController';
+import { SliderHandleBase, HandleValues } from './SliderHandle';
 
 export const variants = ['filled', 'ramp', 'range', 'tick'];
 
-export class Slider extends Focusable {
+const ValueConverter = {
+    fromAttribute: (value: string): number | HandleValues => {
+        const numeric = parseFloat(value);
+        if (!isNaN(numeric)) return numeric;
+        return JSON.parse(value) as HandleValues;
+    },
+    toAttribute: (value: number | HandleValues) => {
+        if (typeof value === 'number') return value.toString();
+        return JSON.stringify(value);
+    },
+};
+export class Slider extends SliderHandleBase {
     public static get styles(): CSSResultArray {
         return [sliderStyles];
     }
 
-    @property()
-    public type = '';
+    private handleController!: HandleController;
 
-    @property({ type: Number, reflect: true })
-    public get value(): number {
-        return this._value;
+    constructor() {
+        super();
+        this.handleController = new HandleController(this);
     }
 
-    public set value(value: number) {
-        const oldValue = this.value;
-        if (this.input) {
-            this.input.value = String(value);
-        }
-        const newValue = this.input ? parseFloat(this.input.value) : value;
+    @property()
+    public type = '';
 
-        if (newValue === oldValue) {
-            return;
+    @property({ reflect: true, converter: ValueConverter })
+    public get value(): number | HandleValues {
+        if (this.handleController.hasExplicitHandles) {
+            return this.handleController.values;
+        } else {
+            return this._value;
         }
+    }
 
-        this._value = newValue;
-        this.requestUpdate('value', oldValue);
+    public set value(value: number | HandleValues) {
```

Comment:
I would wonder if `this.handleController.hasExplicitHandles && this.handleController.count` could handle `isNumber` and if `!this.handleController.hasExplicitHandles` could handle `HandleValues.length === 1`. Some logic is too much...but could be interesting.

Reply:
I changed the value attribute to be simpler